SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityNewTaskChecker.java

```diff
@@ -65,31 +68,35 @@
   private final LoadBalancerClient lbClient;
 
   private final Map<String, Future<?>> taskIdToCheck;
+  private final ConcurrentMap<String, LongAdder> requestIdToUnhealthyCount;
 
   private final ScheduledExecutorService executorService;
 
   private final SingularityAbort abort;
   private final SingularityExceptionNotifier exceptionNotifier;
   private final SingularityDeployHealthHelper deployHealthHelper;
   private final DisasterManager disasterManager;
+  private final SingularityMailer mailer;
 
   @Inject
   public SingularityNewTaskChecker(@Named(SingularityMainModule.NEW_TASK_THREADPOOL_NAME) ScheduledExecutorService executorService, RequestManager requestManager,
                                    SingularityConfiguration configuration, LoadBalancerClient lbClient, TaskManager taskManager, SingularityExceptionNotifier exceptionNotifier, SingularityAbort abort,
-                                   SingularityDeployHealthHelper deployHealthHelper, DisasterManager disasterManager) {
+                                   SingularityDeployHealthHelper deployHealthHelper, DisasterManager disasterManager, SingularityMailer mailer) {
     this.configuration = configuration;
     this.requestManager = requestManager;
     this.taskManager = taskManager;
     this.lbClient = lbClient;
     this.abort = abort;
 
     this.taskIdToCheck = Maps.newConcurrentMap();
+    this.requestIdToUnhealthyCount = Maps.newConcurrentMap();
```

Comment:
documenting from in-person conversation. Right now this would reset when singularity deploys. It would be good to persist these counts somewhere